arangod/Agency/Agent.cpp

```diff
@@ -806,12 +810,28 @@ void Agent::load() {
   }
 }
 
+
+/// Still member of agency?
+bool Agent::legitimate() {
+  legitimacy_t res = _config.legitimise(_configVersion);
+  _configVersion = res.version;
+  return res.legitimate;
+}
+
+
 /// Still leading? Under MUTEX from ::read or ::write
 bool Agent::challengeLeadership() {
+
+  // Still member of the outfit? If not leadership is challenged.
+  // Should be fine to diverge in membership from LOCK below.
+  if (!legitimate()) {
+    return true;
+  }
+
   MUTEX_LOCKER(tiLocker, _tiLock);
-  size_t good = 0;
   
-  std::string const myid = id();
+  size_t good = 0;
+  std::string const myid = id();  
   
   for (auto const& i : _lastAcked) {
     if (i.first != myid) {  // do not count ourselves
```

Comment:
This still counts stuff in _lastAcked from servers which might no longer be active. Or are these entries deleted elsewhere. Note however, that this must all happen only under the _tiLock. No other locks may be acquired!